scylla-cql/src/frame/frame_errors.rs

```diff
@@ -37,6 +37,8 @@ pub enum FrameError {
 pub enum ParseError {
```

Comment:
Please remind me, is the current plan to get rid of ParseError completely?
If not, we need to remember to add non_exhaustive to it, and other errors (like FrameError above).

Reply:
I ultimately would like to get rid of `ParseError`. It will be decoupled into multiple error types, which will be more specific and what's the most important - they will be typed. As of now, most of the `ParseError`s are stringified (e.g. converted to `QueryError::InvalidMessage`).

There is a similar issue with a `FrameError`, however I'm not planning to remove it. I'd like to decouple it and introduce a distinction between `FrameSerializationError` (for client frames) and `FrameDeserializationError` (for server frames). These two could be then represented as variants of `{Query/NewSession}Error` (and not stringified to `QueryError::InvalidMessage`, which is something we currently do).